### PR TITLE
ci: forge fmt over all sol files in testdata

### PR DESCRIPTION
**Pull request by joshieDo**
Dogfooding!

Ran `forge fmt` over our *.sol files, and added a check to the CI.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -201,6 +201,14 @@ jobs:
...
           token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+      
+      - name: forge fmt
+        run: forge fmt --check testdata/
+
```
Comment (mattsse) [inline .github/workflows/test.yml:211]: cool!

do we have this in the book already?

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
@joshieDo need to either reset the fixtures for `abi2solidity*` tests

or plugin formatting when generating interfaces.

or exclude the folder

File: .github/workflows/test.yml (hunk 1 above)
Comment (joshieDo) [reply .github/workflows/test.yml:211]: not yet, there's only an issue https://github.com/foundry-rs/book/issues/38

File: utils/src/lib.rs [hunk 2]
```diff
@@ -729,7 +730,21 @@ pub fn abi_to_solidity(contract_abi: &Abi, mut contract_name: &str) -> Result<St
...
+    let parsed = parse(&sol).map_err(|diags| {
+        eyre::eyre!(
+            "Failed to parse Solidity code for generated code. Leaving source unchanged.\nDebug info: {:?}",
+            diags
+        )
+    })?;
+
+    let mut output = String::new();
+    format(&mut output, parsed, FormatterConfig::default())
+        .map_err(|err| eyre::eyre!(err.to_string()))?;
```
Comment (mattsse) [inline utils/src/lib.rs:745]: lgtm, but looks like we can add a simple function directly in `fmt` that formats a single file?

**Review by mattsse — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -201,6 +201,14 @@ jobs:
           args: --all --all-features -- -D warnings
           token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+      
+      - name: forge fmt
+        run: forge fmt --check testdata/
```
[the remark this answers is not in the document]
Comment (mattsse) [reply .github/workflows/test.yml:210]: right, simple enough, and performant enough to run `cargo run --bin forge -- fmt --check testdata/`

**Review by mattsse — APPROVED** [review]
lgtm

File: cli/src/cmd/forge/fmt.rs [hunk 4]
```diff
@@ -63,7 +63,7 @@ impl FmtArgs {
         let mut paths = self.paths.iter().peekable();
 
         if let Some(path) = paths.peek() {
-            if *path == Path::new("-") || !atty::is(atty::Stream::Stdin) {
+            if *path == Path::new("-") && !atty::is(atty::Stream::Stdin) {
```
Comment (joshieDo) [inline cli/src/cmd/forge/fmt.rs:66]: this way we allow
`cat file.sol | forge fmt -`
`forge fmt - < file.sol`

and it doesn't hang with `forge fmt - `, like it's doing on master

Comment (joshieDo) [reply cli/src/cmd/forge/fmt.rs:66]: @shekhirin does it make sense?

**Review by rkrasiuk — APPROVED** [review]

File: cli/src/cmd/forge/fmt.rs (hunk 4 above)
Comment (shekhirin) [reply cli/src/cmd/forge/fmt.rs:66]: @joshieDo yep, agree!

**Pull request merged**